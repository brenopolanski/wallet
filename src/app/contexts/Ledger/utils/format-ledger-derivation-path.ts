import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
