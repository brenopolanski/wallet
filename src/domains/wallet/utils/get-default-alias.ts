import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
