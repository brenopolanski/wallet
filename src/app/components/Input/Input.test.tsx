import React from "react";
import { render, screen } from "utils/testing-library";

import { Input } from "./Input";

describe("Input", () => {
	it.each([true, false])("should render when isCompact = %s", (isCompact: boolean) => {
		const { asFragment } = render(<Input isCompact={isCompact} />);

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render as custom element", () => {
		const { asFragment } = render(<Input as="select" />);
		const input = screen.getByTestId("Input");

		expect(input.tagName).toBe("SELECT");

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with focus", () => {
		const { asFragment } = render(<Input isFocused />);

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with invalid", () => {
		const { asFragment } = render(<Input isInvalid={true} errorMessage="Field invalid" />);
		const input = screen.getByTestId("Input__error");

		expect(input).toBeVisible();

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with addons", () => {
		const { asFragment } = render(
			<Input
				addons={{
					end: {
						content: <span>end</span>,
						wrapperClassName: "some-class-name",
					},
					start: {
						content: <span>start</span>,
					},
				}}
			/>,
		);

		expect(screen.getByText("end")).toBeInTheDocument();
		expect(screen.getByText("end").parentElement?.parentElement).toHaveClass("some-class-name");

		expect(screen.getByText("start")).toBeInTheDocument();

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with suggestion", () => {
		const { asFragment } = render(<Input suggestion="suggestion" />);

		expect(screen.getByTestId("Input__suggestion")).toBeInTheDocument();

		expect(asFragment()).toMatchSnapshot();
	});
});
