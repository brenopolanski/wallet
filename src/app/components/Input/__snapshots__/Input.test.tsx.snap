// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render as custom element 1`] = `
<DocumentFragment>
  <div
    class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR"
  >
    <div
      class="relative flex flex-1 h-full"
    >
      <select
        class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
        data-testid="Input"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input should render when isCompact = false 1`] = `
<DocumentFragment>
  <div
    class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR"
  >
    <div
      class="relative flex flex-1 h-full"
    >
      <input
        class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
        data-testid="Input"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input should render when isCompact = true 1`] = `
<DocumentFragment>
  <div
    class="Input__InputWrapperStyled-sc-1i8sd1r-0 iIqeeo"
  >
    <div
      class="relative flex flex-1 h-full"
    >
      <input
        class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
        data-testid="Input"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input should render with addons 1`] = `
<DocumentFragment>
  <div
    class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR"
  >
    <span>
      start
    </span>
    <div
      class="relative flex flex-1 h-full"
    >
      <input
        class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
        data-testid="Input"
        value=""
      />
    </div>
    <div
      class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600 some-class-name"
    >
      <div
        class=""
      >
        <span>
          end
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input should render with focus 1`] = `
<DocumentFragment>
  <div
    class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR"
  >
    <div
      class="relative flex flex-1 h-full"
    >
      <input
        class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
        data-testid="Input"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input should render with invalid 1`] = `
<DocumentFragment>
  <div
    class="Input__InputWrapperStyled-sc-1i8sd1r-0 lnWRPe"
  >
    <div
      class="relative flex flex-1 h-full"
    >
      <input
        aria-invalid="true"
        class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
        data-testid="Input"
        value=""
      />
    </div>
    <div
      class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-danger-500"
    >
      <span
        data-errortext="Field invalid"
        data-testid="Input__error"
      >
        <div
          class="sc-bdvvtL qGXhc text-theme-danger-500"
          height="20"
          width="20"
        >
          <svg>
            circle-exclamation-mark.svg
          </svg>
        </div>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input should render with suggestion 1`] = `
<DocumentFragment>
  <div
    class="fixed invisible w-auto whitespace-nowrap"
  >
    â€¦
  </div>
  <div
    class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR"
  >
    <div
      class="relative flex flex-1 h-full"
    >
      <input
        class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
        data-testid="Input"
        value=""
      />
      <span
        class="absolute inset-y-0 flex items-center font-normal opacity-50 pointer-events-none w-full"
        data-testid="Input__suggestion"
      >
        <span
          class="truncate"
        >
          suggestion
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
